IPython/core/display.py

```diff
@@ -119,15 +118,18 @@ def publish_display_data(data, metadata=None, source=None, *, transient=None, **
     transient : dict, keyword-only
         A dictionary of transient data, such as display_id.
         """
+
+    sentinel = object()
+    transient = kwargs.get('transient', sentinel)
+    if transient is sentinel:
+        raise TypeError("publish_display_data() missing 1 required keyword-only argument: 'transient'")
```

Comment:
For backwards compatibility, do we need to set up an empty transient dict and pass it on instead for `display_data`?